cache/clock_cache.cc

```diff
@@ -226,32 +289,70 @@ void ClockHandleTable::Rollback(const Slice& key, uint32_t probe) {
   }
 }
 
+void ClockHandleTable::ClockRun(size_t charge,
+                                autovector<ClockHandle>* deleted) {
+  // TODO(Guido) When an element is in the probe sequence of a
+  // hot element, it will be hard to get an exclusive ref.
+  // We may need a mechanism to avoid that an element sits forever
+  // in cache waiting to be evicted.
+  assert(charge <= capacity_);
+  uint32_t max_iterations = GetTableSize();
+  size_t usage_local = usage_;
+  // TODO(Guido) Evict more than strictly necessary?
+  while (usage_local + charge > capacity_ && max_iterations--) {
+    uint32_t clock_pointer_local = ModTableSize(
+        clock_pointer_++);  // (x mod 2^a) mod 2^b = x mod 2^b for a >= b.
+    ClockHandle* h = &array_[clock_pointer_local];
+
+    if (h->TryExclusiveRef()) {
+      if (h->WillBeDeleted()) {
+        Remove(h, deleted);
```

Comment:
```Remove()``` will update ```charge_``` and ```occupancy_```. If multiple insertions are happening in parallel, it could result in those cache lines thrashing. Would it be better to evict as many as needed and update in one shot?

Reply:
Yes. I'll implement this. ``clock_pointer_`` is also contended among all inserts doing eviction, and we can also increment it once.

An alternative approach is to have a single thread on every shard periodically run the eviction algorithm. This could be future optimization work.

Reply:
I fixed all of this, including the ``clock_pointer_`` thing.